### PR TITLE
feat(rum-core): custom transaction name for user-interactions

### PR DESCRIPTION
**Pull request by David-Development**
+ fix https://github.com/elastic/apm-agent-rum-js/issues/1091
adds support for custom user-interaction transaction names through custom data-attribute as explained here: https://github.com/elastic/apm-agent-rum-js/issues/1091

### CODE REVIEW

**Comment by David-Development** [comment]
@vigneshshanmugam Let me know if I need to do any changes. The lint check fails for some reason but I can't seem to find any details on what the problem is

**Comment by vigneshshanmugam** [comment]
@David-Development Apologies for the delay, I will try to get this in today and we would do a release next week.

File: packages/rum-core/src/performance-monitoring/performance-monitoring.js [hunk 1]
```diff
@@ -188,22 +188,28 @@ export default class PerformanceMonitoring {
...
-          additionalInfo = `["${name}"]`
+        // use custom html attribute 'data-transaction-name' - otherwise fall back to "tagname" + "name"-Attribute
+        let transactionName = ''
+        if (!!target.dataset.transactionName) {
+          transactionName = target.dataset.transactionName
+        } else {
+          const name = target.getAttribute('name')
+          if (!!name) {
+            transactionName = `${tagName}["${name}"]`
+          } else {
```
Comment (vigneshshanmugam) [inline packages/rum-core/src/performance-monitoring/performance-monitoring.js:201]: one small suggestion, we can set the transactionName to tagName during initialization. 

```
let transactionName = tagName

```

**Review by vigneshshanmugam — APPROVED** [review]
Thanks @David-Development, will merge once build passes.

**Pull request merged**